src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_lord_jaraxxus.cpp

```diff
@@ -111,592 +108,459 @@ enum Misc
-                Unit* target = GetTarget();
-                if (caster && target)
-                {
-                    if (target->HasUnitState(UNIT_STATE_CASTING))
-                    {
-                        caster->CastSpell(target, SPELL_MISTRESS_KISS_DAMAGE_SILENCE, true);
-                        target->RemoveAurasDueToSpell(GetSpellInfo()->Id);
-                    }
-                }
-            }
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MISTRESS_KISS_DAMAGE_SILENCE });
+    }
 
-            void Register() override
+    void HandleDummyTick(AuraEffect const* /*aurEff*/)
+    {
+        Unit* target = GetTarget();
+        if (Unit* caster = GetCaster())
+            if (target->HasUnitState(UNIT_STATE_CASTING))
             {
-                OnEffectPeriodic += AuraEffectPeriodicFn(spell_mistress_kiss_AuraScript::HandleDummyTick, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+                caster->CastSpell(target, SPELL_MISTRESS_KISS_DAMAGE_SILENCE, true);
+                target->RemoveAurasDueToSpell(GetSpellInfo()->Id);
             }
-        };
+    }
 
-        AuraScript* GetAuraScript() const override
-        {
-            return new spell_mistress_kiss_AuraScript();
-        }
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_mistress_kiss::HandleDummyTick, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
 };
 
-class MistressKissTargetSelector
+// 66336, 67076, 67077, 67078 - Mistress' Kiss
+class spell_mistress_kiss_area : public SpellScript
 {
-    public:
-        MistressKissTargetSelector() { }
-
-        bool operator()(WorldObject* unit) const
-        {
-            if (unit->GetTypeId() == TYPEID_PLAYER)
-                if (unit->ToPlayer()->getPowerType() == POWER_MANA)
-                    return false;
+    PrepareSpellScript(spell_mistress_kiss_area);
 
-            return true;
-        }
-};
-
-class spell_mistress_kiss_area : public SpellScriptLoader
-{
-    public:
-        spell_mistress_kiss_area() : SpellScriptLoader("spell_mistress_kiss_area") { }
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ static_cast<uint32>(spellInfo->Effects[EFFECT_0].CalcValue()) });
+    }
 
-        class spell_mistress_kiss_area_SpellScript : public SpellScript
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        // get a list of players with mana
+        targets.remove_if([](WorldObject* unit)
         {
-            PrepareSpellScript(spell_mistress_kiss_area_SpellScript);
-
-            void FilterTargets(std::list<WorldObject*>& targets)
-            {
-                // get a list of players with mana
-                targets.remove_if(MistressKissTargetSelector());
-                if (targets.empty())
-                    return;
+            return unit->GetTypeId() == TYPEID_PLAYER && unit->ToPlayer()->getPowerType() == POWER_MANA;
```

Comment:
IsPlayer()

Reply:
Player check is probably useless here anyway, since the spell has ONLY_TARGET_PLAYERS attribute, but good sanity check I guess.

Reply:
yeah, but still can hit corpses i guess (i saw it in sindragosa)
ToPlayer() is not needed there, i will remove xD